model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/clientscope/HotRodClientScopeEntity.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.storage.hotRod.clientscope;
+
+import org.infinispan.protostream.annotations.ProtoDoc;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.keycloak.models.map.annotations.GenerateHotRodEntityImplementation;
+import org.keycloak.models.map.client.MapProtocolMapperEntity;
+import org.keycloak.models.map.clientscope.MapClientScopeEntity;
+import org.keycloak.models.map.storage.hotRod.client.HotRodProtocolMapperEntity;
+import org.keycloak.models.map.storage.hotRod.common.AbstractHotRodEntity;
+import org.keycloak.models.map.storage.hotRod.common.HotRodAttributeEntityNonIndexed;
+import org.keycloak.models.map.storage.hotRod.common.UpdatableHotRodEntityDelegateImpl;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+@GenerateHotRodEntityImplementation(
+        implementInterface = "org.keycloak.models.map.clientscope.MapClientScopeEntity",
+        inherits = "org.keycloak.models.map.storage.hotRod.clientscope.HotRodClientScopeEntity.AbstractHotRodClientScopeEntityDelegate"
+)
+@ProtoDoc("@Indexed")
+public class HotRodClientScopeEntity extends AbstractHotRodEntity  {
+
+    @ProtoField(number = 1, required = true)
+    public int entityVersion = 1;
+
+    @ProtoField(number = 2, required = true)
+    public String id;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 3)
+    public String realmId;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @ProtoField(number = 4)
+    public String name;
+
+    @ProtoField(number = 5)
+    public String protocol;
+
+    @ProtoField(number = 6)
+    public String description;
+
+    @ProtoField(number = 7, collectionImplementation = LinkedHashSet.class)
```

Comment:
Do we need to have this ordered? Why don't we use Set? @vramik Why do we use Collection in [`MapClientScopeEntity`](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/clientscope/MapClientScopeEntity.java#L90)? It seems that the generator chose `LinkedList` as the default collection. Do we need to order scope mappings?

Reply:
There is `Collection` used to be consistent with [`MapClientEntity`](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/client/MapClientEntity.java#L83). I don't think we need ordering here.

Reply:
I put there just `Collection` and removed specific implementation.